### PR TITLE
docs(walkthrough): add 'Stress-testing your solutions' walkthrough (#437)

### LINKED ISSUE #437: docs(walkthrough): add 'Stress-testing your solutions' walkthrough (Authoring track, step 3)
Part of the Walkthrough restructure (see `docs/plans/2026-05-07-walkthrough-restructure-design.md`).

## Scope

New page under **Walkthrough → Authoring a problem**, step 3. Distinct from the existing feature-guide page `setters/stress-testing.md` — that one is reference; this one is a story.

Frame it as: *"my `wa-overflow.cpp` is failing somewhere — find me a tiny counterexample."* Walk through:

1. Defining a finder for the WA solution.
2. Running `rbx stress`.
3. Inspecting the failing input.
4. Promoting the counterexample to a manual test under `manual_tests/` so it sticks across builds.

## Recording

- asciinema: full `rbx stress` cast from kickoff to counterexample.

## Dependency

Depends on the CLI affordance for promoting a stress finding to a manual test (separate issue). If that lands first, the walkthrough uses it; otherwise it documents the manual file-copy workaround.

### PR DESCRIPTION
**Pull request by rsalesc**
Closes #437.

Adds a new narrative walkthrough under **Walkthrough → Authoring a problem**: _"Stress-testing your solutions"_. It continues the sum-of-N problem from First steps and walks the reader through finding a tiny counterexample for `wa-overflow.cpp`'s int32 overflow with `rbx stress`, then making that counterexample stick.

## What's here
- **New page** `docs/setters/stress-testing-walkthrough.md` — story-style, distinct from the `setters/stress-testing.md` reference page (it links there for the operator table, fuzzing, `--slowest`, saved stresses rather than restating them).
- **`mkdocs.yml`** — one nav entry under Authoring a problem, after First steps.
- **`docs/setters/first-steps.md`** — repointed the "Stress test" next-steps card to the new walkthrough.
- **Design + plan docs** under `docs/plans/`.

## Key design decision
Step 4 ("make the counterexample stick") uses the **existing** `rbx stress` add-to-test-group prompt (create `testplan/corner.txt`) rather than blocking on the unbuilt promote-to-`manual_tests/` affordance (#442). The page carries a forward-pointer note, and #442 has a comment to revise this step once that lands.

## Follow-up for the maintainer
The asciinema cast is a **placeholder**: the page contains `{{ asciinema("REPLACE_ME_CAST_ID") }}` with a `<!-- TODO(#437) -->` marker at the "Running the stress" beat. Record the `rbx stress` run (kickoff → counterexample) and substitute the real cast id before publishing. I can't produce the recording.

## Verification
`uv run mkdocs build` succeeds. The technical content (int32-overflow math, generator/finder expressions, and the add-to-test-group flow) was reviewed against source in `rbx/box/cli.py`. Note: `mkdocs build --strict` reports 9 warnings, all pre-existing on `main` and unrelated to this change — the new page introduces zero new warnings and no broken links.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/setters/stress-testing-walkthrough.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+By default the stress runs for about 10 seconds and stops as soon as it finds the first
+match. You can tune both the number of findings and the timeout with `-n` and `-t` — see
+[Stress testing](stress-testing.md) for the details.
+
+## Inspecting the counterexample
+
+When a match is found, `rbx stress` prints a report and shows the exact generator call
+that produced the failing testcase, along with the input itself.
+
+Reading the input is illuminating: it's just a few large numbers whose true sum exceeds
```
Comment (rsalesc) [inline docs/setters/stress-testing-walkthrough.md]: this paragraph is weird: illuminating is a weird and non-natural choice of word, and there is no need for so many details about the testcase.

Comment (rsalesc) [reply docs/setters/stress-testing-walkthrough.md]: Tightened in ff4e7b0 — dropped "illuminating" and the extra testcase detail. Now just: it's a small input whose sum overflows `int32_t`, so wa-overflow.cpp prints the wrong value while main.cpp gets it right.
